### PR TITLE
RNA: Add ActionPopover component

### PR DESCRIPTION
**Pull request by dhasilva**
Part of #27292

#### Changes proposed in this Pull Request:

* Adds the `ActionPopover` component

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Open the Storybook with `( cd projects/js-packages/storybook && pnpm run storybook:dev )`
* Go to `JS Packages/Components/Action Popover`
* Check that it complies with the component needed on the linked issue

TTzDk3dzct551hYWRXPFa3-fi-3062%3A30275&t=l1snOM47GOlP7xA1-4

### CODE REVIEW

**Comment by lhkowalski** [comment]
I can confirm that the action popover looks like this:

Screen Shot 2022-11-29 at 15 29 14

That is what the linked issue is asking for.

File: projects/js-packages/components/components/action-popover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	return (
+		<Popover { ...popoverProps }>
+			<div className={ styles.wrapper }>
+				<div className={ styles.header }>
+					<Text variant="title-small" className={ styles.title }>
+						{ title }
+					</Text>
+					{ ! hideCloseButton && (
+						<button aria-label="close" className={ styles[ 'close-button' ] } onClick={ onClose }>
```
Comment (renatoagds) [inline projects/js-packages/components/components/action-popover/index.tsx]: We shouldn't be using jetpack button? It now support only icon

File: projects/js-packages/components/components/action-popover/stories/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.wrapper {
+	height: 100vh;
+	width: 100%;
```
Comment (renatoagds) [inline projects/js-packages/components/components/action-popover/stories/styles.module.scss]: `100vw`?

File: projects/js-packages/components/components/action-popover/stories/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.wrapper {
```
Comment (renatoagds) [inline projects/js-packages/components/components/action-popover/stories/styles.module.scss:1]: For me, looks strange that we're taking the entire screen and not providing some visual feedback that we're doing it (for example a light dark background that regular is used).

This kind of generate the impression that the page is frozen and do not highlight the action popover.

**Comment by retrofox** [comment]
image

The arrow position looks funny when it isn't centered.

File: projects/js-packages/components/components/action-popover/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,49 @@
+.wrapper {
+	width: 304px;
+	min-height: 230px;
```
Comment (retrofox) [inline projects/js-packages/components/components/action-popover/styles.module.scss:3]: could we define these values based on the `--spacing-base` too?

```suggestion
	width: calc( var( --spacing-base ) * 38 ); // 304px
	min-height: calc( var( --spacing-base ) * 29 ); // almost 230px!;
```

File: projects/js-packages/components/components/action-popover/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	display: flex;
+	flex-direction: column;
+
+	.header {
+		display: flex;
+		margin-bottom: calc( var( --spacing-base ) * 2 ); // 16px
+		justify-content: space-between;
+
+		.title {
+			max-width: 226px;
```
Comment (retrofox) [inline projects/js-packages/components/components/action-popover/styles.module.scss:14]: similar to the comment above

File: projects/js-packages/components/components/action-popover/styles.module.scss (hunk 4 above)
Comment (renatoagds) [reply projects/js-packages/components/components/action-popover/styles.module.scss:3]: don't think we should use `spacing` with `sizes`, these are different things when we talk about design, and if in some moment we decide to update the spacing value we shouldn't affect the sizes.

Comment (retrofox) [reply projects/js-packages/components/components/action-popover/styles.module.scss:3]: yeah, you're right. What I'd like to guarantee is defining the sizes also based on  multiples of eight

Comment (renatoagds) [reply projects/js-packages/components/components/action-popover/styles.module.scss:3]: agree that we should use multiples, it sounds better on readability than random numbers, but maybe we could introduce a new variable or do it manually?

File: projects/js-packages/components/components/action-popover/stories/styles.module.scss (hunk 2 above)
Comment (dhasilva) [reply projects/js-packages/components/components/action-popover/stories/styles.module.scss]: Not necessary. Removed the width, as it is already all the available space.

**Review by retrofox — APPROVED** [review]
LGTM. We can iterate later in case we consider needed

**Pull request merged**